<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Supervized classification of multivariate count table with the Poisson discriminant Analysis • PLNmodels</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Supervized classification of multivariate count table with the Poisson discriminant Analysis">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PLNmodels</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.1003</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a>
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Trichoptera.html">The trichoptera data set</a>
    </li>
    <li>
      <a href="../articles/PLN.html">PLN: Multivariate Poisson regression</a>
    </li>
    <li>
      <a href="../articles/PLNPCA.html">PLNPCA: Dimension Reduction</a>
    </li>
    <li>
      <a href="../articles/PLNnetwork.html">PLNnetwork: sparse structure inference</a>
    </li>
    <li>
      <a href="../articles/PLNLDA.html">PLNLDA: Discriminant Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jchiquet/PLNmodels">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Supervized classification of multivariate count table with the Poisson discriminant Analysis</h1>
                        <h4 class="author">PLN team</h4>
            
            <h4 class="date">2018-12-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jchiquet/PLNmodels/blob/master/vignettes/PLNLDA.Rmd"><code>vignettes/PLNLDA.Rmd</code></a></small>
      <div class="hidden name"><code>PLNLDA.Rmd</code></div>

    </div>

    
    
<div id="preliminaries" class="section level2">
<h2 class="hasAnchor">
<a href="#preliminaries" class="anchor"></a>Preliminaries</h2>
<p>This vignette illustrates the basical use of the <code>PLNLDA</code> function and the methods accompaning the R6 Classes <code>PLNLDA</code> and <code>PLNLDAfit</code>.</p>
<div id="requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h3>
<p>The packages required for the analysis are <strong>PLNmodels</strong> plus some others for data manipulation and representation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tidyverse)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(PLNmodels)</code></pre></div>
</div>
<div id="data-set" class="section level3">
<h3 class="hasAnchor">
<a href="#data-set" class="anchor"></a>Data set</h3>
<p>We illustrate our point with the trichoptera data set, a full description of which can be found in <a href="Trichoptera.html">the corresponding vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(trichoptera)</code></pre></div>
<p>The <code>trichoptera</code> data frame stores a matrix of counts (<code>trichoptera$Abundance</code>), a matrix of offsets (<code>trichoptera$TotalCounts</code>) and some vectors of covariates (<code>trichoptera$Wind</code>, <code>trichoptera$Temperature</code>, etc.) In the following, we’re particularly interested in the <code>trichoptera$Group</code> <strong>discrete</strong> covariate which corresponds to disjoint time spans during which the catching took place. The correspondance between group label and time spans is:</p>
<table class="table">
<thead><tr class="header">
<th align="center">Label</th>
<th align="center">Number of Consecutive Nights</th>
<th align="center">Date</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">1</td>
<td align="center">12</td>
<td align="center">June 59</td>
</tr>
<tr class="even">
<td align="center">2</td>
<td align="center">5</td>
<td align="center">June 59</td>
</tr>
<tr class="odd">
<td align="center">3</td>
<td align="center">5</td>
<td align="center">June 59</td>
</tr>
<tr class="even">
<td align="center">4</td>
<td align="center">4</td>
<td align="center">June 59</td>
</tr>
<tr class="odd">
<td align="center">5</td>
<td align="center">4</td>
<td align="center">July 59</td>
</tr>
<tr class="even">
<td align="center">6</td>
<td align="center">1</td>
<td align="center">June 59</td>
</tr>
<tr class="odd">
<td align="center">7</td>
<td align="center">3</td>
<td align="center">June 60</td>
</tr>
<tr class="even">
<td align="center">8</td>
<td align="center">4</td>
<td align="center">June 60</td>
</tr>
<tr class="odd">
<td align="center">9</td>
<td align="center">5</td>
<td align="center">June 60</td>
</tr>
<tr class="even">
<td align="center">10</td>
<td align="center">4</td>
<td align="center">June 60</td>
</tr>
<tr class="odd">
<td align="center">11</td>
<td align="center">1</td>
<td align="center">June 60</td>
</tr>
<tr class="even">
<td align="center">12</td>
<td align="center">1</td>
<td align="center">July 60</td>
</tr>
</tbody>
</table>
</div>
<div id="mathematical-background" class="section level3">
<h3 class="hasAnchor">
<a href="#mathematical-background" class="anchor"></a>Mathematical background</h3>
<p>In the vein of <span class="citation">Fisher (<a href="#ref-Fi36">1936</a>)</span> and <span class="citation">Rao (<a href="#ref-Rao48">1948</a>)</span>, we introduce a multi-class LDA model for multivariate count data which is a variant of the Poisson Lognormal model of <span class="citation">Aitchison and Ho (<a href="#ref-AiH89">1989</a>)</span> (see <a href="PLN.html">the PLN vignette</a> as a reminder). Indeed, it can viewed as a PLN model with a discrete group structure in the latent gaussian space.</p>
This PLN-LDA model can be written in a hierachical framework where a sample of <span class="math inline">\(p\)</span>-dimensional observation vectors <span class="math inline">\(\mathbf{Y}_i\)</span> is related to some <span class="math inline">\(p\)</span>-dimensional vectors of latent variables <span class="math inline">\(\mathbf{Z}_i\)</span> and a discrete structure with <span class="math inline">\(K\)</span> groups in the following way:
<span class="math display">\[\begin{equation} 
  \begin{array}{rcl}
    \text{group structure }   &amp; \mathbf{\mu}_i = \mu_{g_i} &amp; g_i \in \{1, \dots, K\}  \\
    \text{latent space }   &amp; \mathbf{Z}_i \quad \text{indep.} &amp; \mathbf{Z}_i \sim \mathcal{N}({\boldsymbol\mu}_i, \boldsymbol{\Sigma})  &amp; \\
    \text{observation space } &amp; Y_{ij} | Z_{ij} \quad \text{indep.} &amp; Y_{ij} | Z_{ij} \sim \mathcal{P}\left(\exp\{Z_{ij}\}\right)
  \end{array}
\end{equation}\]</span>
<p>where <span class="math inline">\(g_i\)</span> denotes the group sample <span class="math inline">\(i\)</span> belongs to.</p>
The different parameters <span class="math inline">\({\boldsymbol\mu}_k \in\mathbb{R}^p\)</span> corresponds to the group-specific main effects and the variance matrix <span class="math inline">\(\boldsymbol{\Sigma}\)</span> is shared among groups. An equivalent way of writting this model is the following:
<span class="math display">\[\begin{equation} 
  \begin{array}{rcl}
  \text{latent space } &amp;   \mathbf{Z}_i \sim \mathcal{N}({\boldsymbol\mu}_i,\boldsymbol\Sigma) &amp; \boldsymbol{\mu}_i = \mathbf{g}_i^\top \mathbf{M} \\
  \text{observation space } &amp;  Y_{ij} | Z_{ij} \quad \text{indep.} &amp; Y_{ij} | Z_{ij} \sim \mathcal{P}\left(\exp\{Z_{ij}\}\right),
  \end{array}
\end{equation}\]</span>
<p>where, with a slight abuse of notation, <span class="math inline">\(\mathbf{g}_i\)</span> is a group-indicator vector of length <span class="math inline">\(K\)</span> (<span class="math inline">\(g_{ik} = 1 \Leftrightarrow g_i = k\)</span>) and <span class="math inline">\(\mathbf{M} = [\boldsymbol{\mu}_1^\top, \dots, \boldsymbol{\mu}_K^\top]^\top\)</span> is a <span class="math inline">\(K \times p\)</span> matrix collecting the group-specific main effects.</p>
<div id="covariates-and-offsets" class="section level4">
<h4 class="hasAnchor">
<a href="#covariates-and-offsets" class="anchor"></a>Covariates and offsets</h4>
Just like PLN, PLN-LDA generalizes to a formulation close to a multivariate generalized linear model where the main effect is due to a linear combination of the discrete group structure, <span class="math inline">\(d\)</span> covariates <span class="math inline">\(\mathbf{x}_i\)</span> and a vector <span class="math inline">\(\mathbf{o}_i\)</span> of <span class="math inline">\(p\)</span> offsets in sample <span class="math inline">\(i\)</span>. The latent layer then reads
<span class="math display">\[\begin{equation} 
  \mathbf{Z}_i \sim \mathcal{N}({\mathbf{o}_i + \mathbf{g}_i^\top \mathbf{M} + \mathbf{x}_i^\top\boldsymbol\Theta},\boldsymbol\Sigma)
\end{equation}\]</span>
<p>where <span class="math inline">\(\boldsymbol\Theta\)</span> is a <span class="math inline">\(d\times p\)</span> matrix of regression parameters.</p>
</div>
<div id="prediction" class="section level4">
<h4 class="hasAnchor">
<a href="#prediction" class="anchor"></a>Prediction</h4>
<p>Given:</p>
<ul>
<li>a new observation <span class="math inline">\(\mathbf{Y}\)</span> with associated offset <span class="math inline">\(\mathbf{o}\)</span> and covariates <span class="math inline">\(\mathbf{x}\)</span>
</li>
<li>a model with estimated parameters <span class="math inline">\(\hat{\boldsymbol{\Sigma}}\)</span>, <span class="math inline">\(\hat{\boldsymbol{\Theta}}\)</span>, <span class="math inline">\(\hat{\mathbf{M}}\)</span> and group counts <span class="math inline">\((n_1, \dots, n_K)\)</span>
</li>
</ul>
We can predict the observation’s group using Bayes rule as follows: for <span class="math inline">\(k \in {1, \dots, K}\)</span>, compute
<span class="math display">\[\begin{equation}
\begin{aligned}
f_k(\mathbf{Y}) &amp; = p(\mathbf{Y} | \mathbf{g} = k, \mathbf{o}, \mathbf{x}, \hat{\boldsymbol{\Theta}}, \hat{\boldsymbol{\Sigma}}) \\
                &amp; = \boldsymbol{\Phi}_{PLN}(\mathbf{Y}; \mathbf{o} + \boldsymbol{\mu}_k + \mathbf{x}^\top \hat{\boldsymbol{\Theta}}, \hat{\boldsymbol{\Sigma}}) \\
p_k             &amp; = \frac{n_k}{\sum_{k' = 1}^K n_{k'}}    
\end{aligned}
\end{equation}\]</span>
<p>where <span class="math inline">\(\boldsymbol{\Phi}_{PLN}(\bullet; \boldsymbol{\mu}, \boldsymbol{\Sigma})\)</span> is the density function of a PLN distribution with parameters <span class="math inline">\((\boldsymbol{\mu}, \boldsymbol{\Sigma})\)</span>. <span class="math inline">\(f_k(\mathbf{Y})\)</span> and <span class="math inline">\(p_k\)</span> are respectively plug-in estimates of (i) the probability of observing counts <span class="math inline">\(\mathbf{Y}\)</span> in a sample from group <span class="math inline">\(k\)</span> and (ii) the probability that a sample originates from group <span class="math inline">\(k\)</span>.</p>
The posterior probability <span class="math inline">\(\hat{\pi}_k(\mathbf{Y})\)</span> that observation <span class="math inline">\(\mathbf{Y}\)</span> belongs to group <span class="math inline">\(k\)</span> and most likely group <span class="math inline">\(\hat{k}(\mathbf{Y})\)</span> can thus be defined as
<span class="math display">\[\begin{equation}
\begin{aligned}
\hat{\pi}_k(\mathbf{Y}) &amp; = \frac{p_k f_k(\mathbf{Y})}{\sum_{k' = 1}^K p_{k'} f_{k'}(\mathbf{Y})} \\
\hat{k}(\mathbf{Y})     &amp; = \underset{k \in \{1, \dots, K\}}{\arg\max} \hat{\pi}_k(\mathbf{Y})
\end{aligned}
\end{equation}\]</span>
</div>
<div id="optimization-by-variational-inference" class="section level4">
<h4 class="hasAnchor">
<a href="#optimization-by-variational-inference" class="anchor"></a>Optimization by Variational inference</h4>
<p>Classification and prediction are the main objectives in (PLN-)LDA. To reach this goal, we first need to estimate the model parameters. Inference in PLN-LDA focuses on the group-specific main effects <span class="math inline">\(\mathbf{M}\)</span>, the regression parameters <span class="math inline">\(\boldsymbol\Theta\)</span> and the covariance matrix <span class="math inline">\(\boldsymbol\Sigma\)</span>. Technically speaking, we can treat <span class="math inline">\(\mathbf{g}_i\)</span> as a discrete covariate and estimate <span class="math inline">\([\mathbf{M}, \boldsymbol{\Theta}]\)</span> using the same strategy as for the standard <strong>PLN</strong> model. Briefly, we adopt a variational strategy to approximate the log-likelihood function and optimize the consecutive variational surrogate of the log-likelihood with a gradient-ascent-based approach. To this end, we rely on the CCSA algorithm of <span class="citation">Svanberg (<a href="#ref-Svan02">2002</a>)</span> implemented in the C++ library <span class="citation">(Johnson <a href="#ref-nlopt">2011</a>)</span>, which we link to the package.</p>
</div>
</div>
</div>
<div id="analysis-of-trichoptera-data-with-a-pln-lda-model" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-trichoptera-data-with-a-pln-lda-model" class="anchor"></a>Analysis of trichoptera data with a PLN-LDA model</h2>
<p>In the package, the PLN-LDA model is adjusted with the function <code>PLNLDA</code>, which we review in this section. This function adjusts the model and stores it in a object of class <code>PLNLDAfit</code> which inherits from the class <code>PLNfit</code>, so we strongly recommend the reader to be somehow comfortable with <code>PLN</code> and <code>PLNfit</code> before using <code>PLNLDA</code> (see <a href="PLN.html">the PLN vignette</a>).</p>
<div id="a-model-with-main-effects-and-no-covariates" class="section level3">
<h3 class="hasAnchor">
<a href="#a-model-with-main-effects-and-no-covariates" class="anchor"></a>A model with main effects and no covariates</h3>
<p>We start by adjusting the above model to the Trichoptera data set. We use <code>Group</code>, the catching time spans, as a discrete structure and use log as an offset to capture differences in sampling luck.</p>
<p>The model can be fitted with the function <code>PLNLDA</code> as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_nocov &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PLNLDA.html">PLNLDA</a></span>(Abundance <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/offset">offset</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log</a></span>(TotalCounts)),
                      <span class="dt">grouping =</span> trichoptera<span class="op">$</span>Group, 
                      <span class="dt">data =</span> trichoptera)</code></pre></div>
<pre><code>## 
##  Performing discriminant Analysis...
##  DONE!</code></pre>
<p>Note that <code>PLNLDA</code> uses the standard <code>formula</code> interface, like every other model in the <strong>PLNmodels</strong> package.</p>
<div id="structure-of-plnldafit" class="section level4">
<h4 class="hasAnchor">
<a href="#structure-of-plnldafit" class="anchor"></a>Structure of <code>PLNLDAfit</code>
</h4>
<p>The <code>myLDA_nocov</code> variable is an <code>R6</code> object with class <code>PLNLDAfit</code>, which comes with a couple of methods. The most basic is the <code>show/print</code> method, which sends a brief summary of the estimation process and available methods:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_nocov</code></pre></div>
<pre><code>## Linear Discriminant Analysis for Poisson Lognormal distribution
## ==================================================================
##  degrees_freedom    loglik       BIC       ICL R_squared
##              391 -799.5061 -1560.357 -1119.586 0.4795375
## ==================================================================
## * Useful fields 
##     $model_par, $latent, $var_par, $optim_par 
##     $loglik, $BIC, $ICL, $loglik_vec, $degrees_freedom, $criteria 
## * Useful S3 methods
##     print(), coef(), vcov(), fitted(), standard_error(), fisher() 
## * Additional fields for LDA
##     $percent_var, $corr_map, $scores, $group_means 
## * Additional S3 methods for LDA
##     plot.PLNLDAfit(), predict.PLNLDAfit()</code></pre>
<p>Comprehensive information about <code>PLNLDAfit</code> is available via <code><a href="../reference/PLNLDAfit.html">?PLNLDAfit</a></code>.</p>
</div>
<div id="specific-fields" class="section level4">
<h4 class="hasAnchor">
<a href="#specific-fields" class="anchor"></a>Specific fields</h4>
<p>The user can easily access several fields of the <code>PLNLDAfit</code> object using <code>S3</code> methods, the most interesting ones are</p>
<ul>
<li>the <span class="math inline">\(p \times p\)</span> covariance matrix <span class="math inline">\(\hat{\boldsymbol{\Sigma}}\)</span>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/vcov">vcov</a></span>(myLDA_nocov) <span class="op">%&gt;%</span><span class="st"> </span>corrplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/corrplot/topics/corrplot">corrplot</a></span>(<span class="dt">is.corr =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="PLNLDA_files/figure-html/vcov-1.png" width="700"></p>
<ul>
<li>the regression coefficient matrix <span class="math inline">\(\hat{\boldsymbol{\Theta}}\)</span> (in this case <code>NULL</code> as there are no covariates)</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/coef">coef</a></span>(myLDA_nocov)</code></pre></div>
<pre><code>## NULL</code></pre>
<ul>
<li>the <span class="math inline">\(p \times K\)</span> matrix of group means <span class="math inline">\(\mathbf{M}\)</span> (only 8 first groups heres)</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_nocov<span class="op">$</span>group_means[, <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>() <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">grouping1</th>
<th align="right">grouping2</th>
<th align="right">grouping3</th>
<th align="right">grouping4</th>
<th align="right">grouping5</th>
<th align="right">grouping6</th>
<th align="right">grouping7</th>
<th align="right">grouping8</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">-26.04</td>
<td align="right">-6.82</td>
<td align="right">-25.61</td>
<td align="right">-24.43</td>
<td align="right">-27.38</td>
<td align="right">-27.97</td>
<td align="right">-26.28</td>
<td align="right">-25.42</td>
</tr>
<tr class="even">
<td align="right">-26.01</td>
<td align="right">-26.76</td>
<td align="right">-5.66</td>
<td align="right">-24.40</td>
<td align="right">-7.43</td>
<td align="right">-27.94</td>
<td align="right">-26.25</td>
<td align="right">-25.39</td>
</tr>
<tr class="odd">
<td align="right">-2.38</td>
<td align="right">-3.96</td>
<td align="right">-2.36</td>
<td align="right">-2.92</td>
<td align="right">-6.51</td>
<td align="right">-5.54</td>
<td align="right">-2.69</td>
<td align="right">-2.03</td>
</tr>
<tr class="even">
<td align="right">-26.04</td>
<td align="right">-26.80</td>
<td align="right">-5.70</td>
<td align="right">-4.53</td>
<td align="right">-27.32</td>
<td align="right">-28.02</td>
<td align="right">-26.30</td>
<td align="right">-5.52</td>
</tr>
<tr class="odd">
<td align="right">-0.28</td>
<td align="right">-0.26</td>
<td align="right">-0.62</td>
<td align="right">-1.09</td>
<td align="right">-0.61</td>
<td align="right">-0.11</td>
<td align="right">-0.66</td>
<td align="right">-0.80</td>
</tr>
<tr class="even">
<td align="right">-3.99</td>
<td align="right">-3.32</td>
<td align="right">-2.93</td>
<td align="right">-4.47</td>
<td align="right">-6.72</td>
<td align="right">-8.00</td>
<td align="right">-2.85</td>
<td align="right">-3.06</td>
</tr>
</tbody>
</table>
<p>The <code>PLNLDAfit</code> class also benefits from two important methods: <code>plot</code> and <code>predict</code>.</p>
</div>
<div id="plot-method" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-method" class="anchor"></a><code>plot</code> method</h4>
<p>The <code>plot</code> methods provides easy to interpret graphics which reveals here that the groups are well separated:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(myLDA_nocov)</code></pre></div>
<p><img src="PLNLDA_files/figure-html/plot_model-1.png" width="672"></p>
<p>By default, <code>plot</code> shows the first 3 axis of the LDA when there are 4 or more groups and uses special representations for the edge cases of 3 or less groups.</p>
<p><code>ggplot2</code>-savvy users who want to make their own representations can extracts the <span class="math inline">\(n \times (K-1)\)</span> matrix of sample scores from the <code>PLNLDAfit</code> object (only first 8 axes shown here) …</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_nocov<span class="op">$</span>scores[, <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>] <span class="op">%&gt;%</span><span class="st"> </span>head <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table class="table"><tbody>
<tr class="odd">
<td align="right">4291830</td>
<td align="right">-122308.2</td>
<td align="right">-34037.96</td>
<td align="right">-94068.13</td>
<td align="right">29139.20</td>
<td align="right">14574.81</td>
<td align="right">-433.93</td>
<td align="right">-2139.64</td>
</tr>
<tr class="even">
<td align="right">4291976</td>
<td align="right">-127109.7</td>
<td align="right">-33792.73</td>
<td align="right">-94574.86</td>
<td align="right">28957.81</td>
<td align="right">14591.55</td>
<td align="right">-452.62</td>
<td align="right">-2147.13</td>
</tr>
<tr class="odd">
<td align="right">4294557</td>
<td align="right">-171580.3</td>
<td align="right">-31153.01</td>
<td align="right">-98007.39</td>
<td align="right">27378.60</td>
<td align="right">14356.28</td>
<td align="right">-78.99</td>
<td align="right">-2024.78</td>
</tr>
<tr class="even">
<td align="right">4297254</td>
<td align="right">-221949.5</td>
<td align="right">-28199.00</td>
<td align="right">-101709.18</td>
<td align="right">25624.09</td>
<td align="right">13992.65</td>
<td align="right">512.20</td>
<td align="right">-1806.95</td>
</tr>
<tr class="odd">
<td align="right">4295606</td>
<td align="right">-192804.9</td>
<td align="right">-29942.26</td>
<td align="right">-99502.91</td>
<td align="right">26661.80</td>
<td align="right">14157.19</td>
<td align="right">241.20</td>
<td align="right">-1895.95</td>
</tr>
<tr class="even">
<td align="right">4293565</td>
<td align="right">-155859.4</td>
<td align="right">-32114.13</td>
<td align="right">-96742.94</td>
<td align="right">27950.04</td>
<td align="right">14400.85</td>
<td align="right">-145.75</td>
<td align="right">-2036.31</td>
</tr>
</tbody></table>
<p>…or the <span class="math inline">\(p \times (K-1)\)</span> matrix of correlations between scores and (latent) variables</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_nocov<span class="op">$</span>corr_map <span class="op">%&gt;%</span><span class="st"> </span>head <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table class="table"><tbody>
<tr class="odd">
<td align="left">Che</td>
<td align="right">-0.29</td>
<td align="right">0.24</td>
<td align="right">0.53</td>
<td align="right">0.20</td>
<td align="right">-0.62</td>
<td align="right">-0.19</td>
<td align="right">-0.38</td>
<td align="right">-0.14</td>
<td align="right">0.87</td>
<td align="right">-0.43</td>
<td align="right">0.03</td>
</tr>
<tr class="even">
<td align="left">Hyc</td>
<td align="right">-0.07</td>
<td align="right">-0.46</td>
<td align="right">0.07</td>
<td align="right">0.41</td>
<td align="right">-0.04</td>
<td align="right">-0.60</td>
<td align="right">0.83</td>
<td align="right">0.50</td>
<td align="right">-0.11</td>
<td align="right">-0.30</td>
<td align="right">0.57</td>
</tr>
<tr class="odd">
<td align="left">Hym</td>
<td align="right">0.07</td>
<td align="right">-0.18</td>
<td align="right">-0.18</td>
<td align="right">-0.76</td>
<td align="right">0.34</td>
<td align="right">-0.14</td>
<td align="right">-0.29</td>
<td align="right">0.16</td>
<td align="right">-0.30</td>
<td align="right">0.04</td>
<td align="right">-0.63</td>
</tr>
<tr class="even">
<td align="left">Hys</td>
<td align="right">-0.50</td>
<td align="right">0.15</td>
<td align="right">-0.56</td>
<td align="right">-0.05</td>
<td align="right">0.26</td>
<td align="right">-0.43</td>
<td align="right">-0.36</td>
<td align="right">0.13</td>
<td align="right">-0.26</td>
<td align="right">-0.07</td>
<td align="right">0.21</td>
</tr>
<tr class="odd">
<td align="left">Psy</td>
<td align="right">0.26</td>
<td align="right">0.26</td>
<td align="right">0.21</td>
<td align="right">-0.27</td>
<td align="right">-0.35</td>
<td align="right">0.30</td>
<td align="right">-0.15</td>
<td align="right">0.23</td>
<td align="right">0.48</td>
<td align="right">-0.52</td>
<td align="right">-0.35</td>
</tr>
<tr class="even">
<td align="left">Aga</td>
<td align="right">0.10</td>
<td align="right">-0.15</td>
<td align="right">-0.20</td>
<td align="right">-0.89</td>
<td align="right">0.04</td>
<td align="right">-0.15</td>
<td align="right">-0.24</td>
<td align="right">-0.09</td>
<td align="right">-0.08</td>
<td align="right">-0.17</td>
<td align="right">-0.53</td>
</tr>
</tbody></table>
</div>
<div id="predict-method" class="section level4">
<h4 class="hasAnchor">
<a href="#predict-method" class="anchor"></a><code>predict</code> method</h4>
<p>The <code>predict</code> method has a slightly different behavior than its siblings in other models of the <strong>PLNmodels</strong>. The goal of <code>predict</code> is to predict the discrete class based on observed <em>species counts</em> (rather than predicting counts from known covariates).</p>
<p>By default, the <code>predict</code> use the argument <code>type = "posterior"</code> to output the matrix of log-posterior probabilities <span class="math inline">\(\log(\hat{\pi})_k\)</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">predicted.class &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/predict">predict</a></span>(myLDA_nocov, <span class="dt">newdata =</span> trichoptera)
## equivalent to 
## predicted.class &lt;- predict(myLDA_nocov, newdata = trichoptera,  type = "posterior")
predicted.class <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>() <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
<th align="right">6</th>
<th align="right">7</th>
<th align="right">8</th>
<th align="right">9</th>
<th align="right">10</th>
<th align="right">11</th>
<th align="right">12</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">-17.45</td>
<td align="right">-45.54</td>
<td align="right">-42.59</td>
<td align="right">-25.47</td>
<td align="right">-53.73</td>
<td align="right">-32.77</td>
<td align="right">-25.91</td>
<td align="right">-84.66</td>
<td align="right">-81.72</td>
<td align="right">-20.59</td>
<td align="right">-199.25</td>
<td align="right">-72.97</td>
</tr>
<tr class="even">
<td align="right">-9.66</td>
<td align="right">-17.04</td>
<td align="right">-58.04</td>
<td align="right">-21.33</td>
<td align="right">-70.28</td>
<td align="right">-27.31</td>
<td align="right">-14.69</td>
<td align="right">-15.58</td>
<td align="right">-15.20</td>
<td align="right">-14.69</td>
<td align="right">-117.18</td>
<td align="right">-64.53</td>
</tr>
<tr class="odd">
<td align="right">-12.94</td>
<td align="right">-22.56</td>
<td align="right">-111.37</td>
<td align="right">-39.95</td>
<td align="right">-116.19</td>
<td align="right">-30.79</td>
<td align="right">-22.54</td>
<td align="right">-49.15</td>
<td align="right">-43.55</td>
<td align="right">-22.94</td>
<td align="right">-143.43</td>
<td align="right">-116.82</td>
</tr>
<tr class="even">
<td align="right">-18.42</td>
<td align="right">-31.04</td>
<td align="right">-119.21</td>
<td align="right">-115.57</td>
<td align="right">-141.19</td>
<td align="right">-54.03</td>
<td align="right">-41.96</td>
<td align="right">-111.86</td>
<td align="right">-100.47</td>
<td align="right">-44.67</td>
<td align="right">-301.59</td>
<td align="right">-220.90</td>
</tr>
<tr class="odd">
<td align="right">-13.84</td>
<td align="right">-20.00</td>
<td align="right">-51.79</td>
<td align="right">-60.84</td>
<td align="right">-71.71</td>
<td align="right">-38.18</td>
<td align="right">-26.90</td>
<td align="right">-50.91</td>
<td align="right">-44.34</td>
<td align="right">-25.88</td>
<td align="right">-189.13</td>
<td align="right">-112.82</td>
</tr>
<tr class="even">
<td align="right">-9.24</td>
<td align="right">-13.48</td>
<td align="right">-36.28</td>
<td align="right">-24.39</td>
<td align="right">-47.22</td>
<td align="right">-23.74</td>
<td align="right">-14.61</td>
<td align="right">-15.91</td>
<td align="right">-15.29</td>
<td align="right">-14.32</td>
<td align="right">-97.32</td>
<td align="right">-65.86</td>
</tr>
</tbody>
</table>
<p>Setting <code>type = "response"</code>, we can predict the most likely group <span class="math inline">\(\hat{k}\)</span> instead:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">predicted.class &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/predict">predict</a></span>(myLDA_nocov, <span class="dt">newdata =</span> trichoptera,  <span class="dt">type =</span> <span class="st">"response"</span>)
predicted.class</code></pre></div>
<pre><code>##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 
##  1  1  1  1  1  1  9  2  1  1  1  1  2  3  2  2  2  3  3  3  9  3  4  1  4 
## 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 
##  4 12  5  4  5  6  7  7  7  8  8  8  8  8  1  9  9  9 10 10 10 10 11 12 
## Levels: 1 2 3 4 5 6 7 8 9 10 11 12</code></pre>
<p>Finally, we can assess that the predictions are quite similar to the real group (<em>this is not a proper validation of the method as we used dataset for both model fitting and prediction and are thus at risk of overfitting</em>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(predicted.class, trichoptera<span class="op">$</span>Group, <span class="dt">dnn =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"predicted"</span>, <span class="st">"true"</span>))</code></pre></div>
<pre><code>##          true
## predicted  1  2  3  4  5  6  7  8  9 10 11 12
##        1  10  0  0  1  0  0  0  0  1  0  0  0
##        2   1  4  0  0  0  0  0  0  0  0  0  0
##        3   0  1  4  0  0  0  0  0  0  0  0  0
##        4   0  0  0  3  1  0  0  0  0  0  0  0
##        5   0  0  0  0  2  0  0  0  0  0  0  0
##        6   0  0  0  0  0  1  0  0  0  0  0  0
##        7   0  0  0  0  0  0  3  0  0  0  0  0
##        8   0  0  0  0  0  0  0  4  1  0  0  0
##        9   1  0  1  0  0  0  0  0  3  0  0  0
##        10  0  0  0  0  0  0  0  0  0  4  0  0
##        11  0  0  0  0  0  0  0  0  0  0  1  0
##        12  0  0  0  0  1  0  0  0  0  0  0  1</code></pre>
</div>
</div>
<div id="a-model-with-latent-main-effects-and-meterological-covariates" class="section level3">
<h3 class="hasAnchor">
<a href="#a-model-with-latent-main-effects-and-meterological-covariates" class="anchor"></a>A model with latent main effects and meterological covariates</h3>
<p>It is possible to correct for other covariates before finding the LDA axes that best separate well the groups. In our case ,we’re going to use <code>Wind</code> as a covariate and illustrate the main differences with before :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_cov &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PLNLDA.html">PLNLDA</a></span>(Abundance <span class="op">~</span><span class="st"> </span>Wind <span class="op">+</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/offset">offset</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log</a></span>(TotalCounts)), <span class="dt">grouping =</span> trichoptera<span class="op">$</span>Group, <span class="dt">data =</span> trichoptera)</code></pre></div>
<pre><code>## 
##  Performing discriminant Analysis...
##  DONE!</code></pre>
<div id="specific-fields-1" class="section level4">
<h4 class="hasAnchor">
<a href="#specific-fields-1" class="anchor"></a>Specific fields</h4>
<p>All fields of our new <code>PLNLDA</code> fit can be accessed as before with similar results. The only important difference is the result of <code>coef</code>: since we included a covariate in the model, <code>coef</code> now returns a 1-column matrix for <span class="math inline">\(\hat{\boldsymbol{\Theta}}\)</span> instead of <code>NULL</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/coef">coef</a></span>(myLDA_cov) <span class="op">%&gt;%</span><span class="st"> </span>head <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>()</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Wind</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Che</td>
<td align="right">-0.3297942</td>
</tr>
<tr class="even">
<td>Hyc</td>
<td align="right">1.4102419</td>
</tr>
<tr class="odd">
<td>Hym</td>
<td align="right">-0.1927923</td>
</tr>
<tr class="even">
<td>Hys</td>
<td align="right">-0.4547067</td>
</tr>
<tr class="odd">
<td>Psy</td>
<td align="right">0.0368501</td>
</tr>
<tr class="even">
<td>Aga</td>
<td align="right">-0.0611272</td>
</tr>
</tbody>
</table>
<p>The group-specific main effects can still be accessed with <code>$group_means</code> (only first 8 shown here)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myLDA_cov<span class="op">$</span>group_means <span class="op">%&gt;%</span><span class="st"> </span>head <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">grouping1</th>
<th align="right">grouping2</th>
<th align="right">grouping3</th>
<th align="right">grouping4</th>
<th align="right">grouping5</th>
<th align="right">grouping6</th>
<th align="right">grouping7</th>
<th align="right">grouping8</th>
<th align="right">grouping9</th>
<th align="right">grouping10</th>
<th align="right">grouping11</th>
<th align="right">grouping12</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">-16.47</td>
<td align="right">-9.98</td>
<td align="right">-19.62</td>
<td align="right">-15.31</td>
<td align="right">-20.88</td>
<td align="right">-35.66</td>
<td align="right">-24.28</td>
<td align="right">-22.86</td>
<td align="right">-15.76</td>
<td align="right">2.50</td>
<td align="right">8.65</td>
<td align="right">-19.16</td>
</tr>
<tr class="even">
<td align="right">-19.95</td>
<td align="right">-36.58</td>
<td align="right">-2.41</td>
<td align="right">-12.75</td>
<td align="right">-11.24</td>
<td align="right">-40.18</td>
<td align="right">-27.66</td>
<td align="right">-24.66</td>
<td align="right">-11.56</td>
<td align="right">-13.99</td>
<td align="right">-3.79</td>
<td align="right">0.99</td>
</tr>
<tr class="odd">
<td align="right">-1.18</td>
<td align="right">-4.42</td>
<td align="right">-1.77</td>
<td align="right">-1.80</td>
<td align="right">-5.81</td>
<td align="right">-6.85</td>
<td align="right">-2.66</td>
<td align="right">-1.84</td>
<td align="right">-1.58</td>
<td align="right">-1.57</td>
<td align="right">-20.17</td>
<td align="right">-3.31</td>
</tr>
<tr class="even">
<td align="right">-19.87</td>
<td align="right">-26.93</td>
<td align="right">-3.32</td>
<td align="right">-0.48</td>
<td align="right">-22.09</td>
<td align="right">-31.41</td>
<td align="right">-24.47</td>
<td align="right">-5.20</td>
<td align="right">-1.25</td>
<td align="right">-0.65</td>
<td align="right">-15.23</td>
<td align="right">-20.83</td>
</tr>
<tr class="odd">
<td align="right">-0.05</td>
<td align="right">-0.42</td>
<td align="right">-0.47</td>
<td align="right">-0.85</td>
<td align="right">-0.44</td>
<td align="right">-0.43</td>
<td align="right">-0.59</td>
<td align="right">-0.71</td>
<td align="right">-0.06</td>
<td align="right">-0.15</td>
<td align="right">-0.19</td>
<td align="right">-1.03</td>
</tr>
<tr class="even">
<td align="right">-1.89</td>
<td align="right">-4.07</td>
<td align="right">-1.76</td>
<td align="right">-2.47</td>
<td align="right">-5.52</td>
<td align="right">-10.44</td>
<td align="right">-2.71</td>
<td align="right">-2.75</td>
<td align="right">-3.38</td>
<td align="right">-1.83</td>
<td align="right">-18.60</td>
<td align="right">-21.84</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-method-1" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-method-1" class="anchor"></a><code>plot</code> method</h4>
<p>Once again, the <code>plot</code> method is very useful to get a quick look at the results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(myLDA_cov)</code></pre></div>
<p><img src="PLNLDA_files/figure-html/plot_model_cov-1.png" width="672"></p>
</div>
<div id="predict-method-1" class="section level4">
<h4 class="hasAnchor">
<a href="#predict-method-1" class="anchor"></a><code>predict</code> method</h4>
<p>We can again predict the most likely group for each sample :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">predicted.class_cov &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/predict">predict</a></span>(myLDA_cov, <span class="dt">newdata =</span> trichoptera, <span class="dt">type =</span> <span class="st">"response"</span>)</code></pre></div>
<p>and check that we recover the correct class in most cases (again, we used the same dataset for model fitting and group prediction only for ease of exposition):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(predicted.class_cov, trichoptera<span class="op">$</span>Group, <span class="dt">dnn =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"predicted"</span>, <span class="st">"true"</span>))</code></pre></div>
<pre><code>##          true
## predicted  1  2  3  4  5  6  7  8  9 10 11 12
##        1  11  0  0  1  0  0  0  0  1  0  0  0
##        2   1  4  0  0  0  0  0  0  0  0  0  0
##        3   0  1  4  0  0  0  0  0  0  0  0  0
##        4   0  0  0  3  1  0  0  0  0  0  0  0
##        5   0  0  0  0  2  0  0  0  0  0  0  0
##        6   0  0  0  0  0  1  0  0  0  0  0  0
##        7   0  0  0  0  0  0  3  0  0  0  0  0
##        8   0  0  0  0  0  0  0  4  1  0  0  0
##        9   0  0  1  0  0  0  0  0  3  0  0  0
##        10  0  0  0  0  0  0  0  0  0  4  0  0
##        11  0  0  0  0  0  0  0  0  0  0  1  0
##        12  0  0  0  0  1  0  0  0  0  0  0  1</code></pre>
<div id="refs" class="references">
<div id="ref-AiH89">
<p>Aitchison, J., and C.H. Ho. 1989. “The Multivariate Poisson-Log Normal Distribution.” <em>Biometrika</em> 76 (4). Oxford University Press: 643–53.</p>
</div>
<div id="ref-Fi36">
<p>Fisher, R. A. 1936. “The Use of Multiple Measurements in Taxonomic Problems.” <em>Annals of Eugenics</em> 7 (2): 179–88.</p>
</div>
<div id="ref-nlopt">
<p>Johnson, Steven G. 2011. <em>The Nlopt Nonlinear-Optimization Package</em>. <a href="http://ab-initio.mit.edu/nlopt" class="uri">http://ab-initio.mit.edu/nlopt</a>.</p>
</div>
<div id="ref-Rao48">
<p>Rao, C. Radhakrishna. 1948. “The Utilization of Multiple Measurements in Problems of Biological Classification.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 10 (2). [Royal Statistical Society, Wiley]: 159–203.</p>
</div>
<div id="ref-Svan02">
<p>Svanberg, Krister. 2002. “A Class of Globally Convergent Optimization Methods Based on Conservative Convex Separable Approximations.” <em>SIAM Journal on Optimization</em> 12 (2). SIAM: 555–73.</p>
</div>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#preliminaries">Preliminaries</a></li>
      <li><a href="#analysis-of-trichoptera-data-with-a-pln-lda-model">Analysis of trichoptera data with a PLN-LDA model</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Julien Chiquet, Mahendra Mariadassou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
